src/foam/mlang/expr/DateTimeGroupingExpr.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.expr',
+  name: 'DateTimeGroupingExpr',
+  extends: 'foam.mlang.AbstractExpr',
+
+  implements: [ 'foam.core.Serializable' ],
+
+  documentation: `
+    An expr whose value is the name of a date grouping.
+    Currently works with hours, days, weeks, months, and years.
+  `,
+
+  javaImports: [
+    'foam.nanos.auth.CreatedAware',
+    'java.util.Arrays',
+    'java.util.Comparator',
+    'java.util.Date'
+  ],
+
+  properties: [
+    {
+      class: 'Enum',
+      of: 'foam.mlang.expr.DateTypes',
```

Comment:
@lchanmann 
Another duplication 
foam.mlang.expr.DateTypes and 
foam.time.TimeUnit 
What dependencies do we have on expr.DateTypes?